### DISCUSSION TITLE
How to store the results of the calculate_hundegger_machine_data(elements, hundeggertype) method in a variable?

### DISCUSSION THREAD

**Discussion by lucas-ita** [cwapi3d/cwapi3dpython#137] [General]
Hi,

Would it be possible to store the results of the `calculate_hundegger_machine_data(elements, hunderggertype`) method in a variable?
By performing this calculation and accessing the "individual control" option in CadWork, I can obtain information about all elements that come into contact or collide with a specific part. The image below illustrates this process:

![image](https://github.com/cwapi3d/cwapi3dpython/assets/123772761/9bc4fb2c-2098-474a-a4f8-25f118089de9)

My goal is to use the mentioned calculation method and then obtain the geometries that already contain detailed information about collisions and touches. Later, I plan to group each element with all the pieces that come into contact or collide in a scene.

In summary I would like to take advantage of the existing functionality in Cadwork with the `calculate_hundegger_machine_data(elements, hunderggertype)` method. If this is not possible, I am aware that it would be necessary to create this logic using the `check_if_elements_are_in_contact(first_element_id, second_element_id)` and `check_if_elements_are_in_collision(first_element_id, second_element_id)` methods.

**Comment by roehligas** [comment]
I recommend building a temporary dummy-beam with the same raw-sizes and same position as the actual element and then simply substract the original element.

You'll be left with the desired geometries that are the difference between the real and the rough geometry.
Note that in order to have all you'll have to append the ID of the dummy-beam to the returned list of the substract function, it only returns newly generated elements.

This is where the tricky part starts, which is to classify the leftover geometry as operations etc.

I didn't figure that part out yet, but for mere collision control the suggestion above works beautifully for me, we use it in several scripts.

**Comment by Brunner246** [comment]
There will be `delete_processes_keep_cutting_bodies` in the next release
